src/boot/efi/efi-string.c

```diff
@@ -5,7 +5,7 @@
 
 #include "efi-string.h"
 
-#ifdef SD_BOOT
+#if SD_BOOT
```

Comment:
This commit fails to touch all places where we use `#ifdef SD_BOOT`, for example `src/fundamental/memory-util-fundamental.h` still uses it and it causes build fails when playing with this locally:

```
...
[140/2056] Compiling C object src/machine/libmachine-core.a.p/machined-core.c.o
FAILED: src/machine/libmachine-core.a.p/machined-core.c.o 
cc -Isrc/machine/libmachine-core.a.p -Isrc/machine -I../src/machine -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I. -I.. -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/shared -I../src/shared -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu11 -O0 -g -Wno-missing-field-initializers -Wno-unused-parameter -Wdate-time -Wendif-labels -Werror=format=2 -Werror=format-signedness -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=int-conversion -Werror=overflow -Werror=override-init -Werror=return-type -Werror=shift-count-overflow -Werror=shift-overflow=2 -Werror=undef -Wfloat-equal -Wimplicit-fallthrough=5 -Winit-self -Wlogical-op -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wsuggest-attribute=noreturn -Wunused-function -Wwrite-strings -Wno-unused-result -Werror=missing-declarations -Werror=missing-prototypes -fdiagnostics-show-option -fno-common -fno-strict-aliasing -fstack-protector -fstack-protector-strong -fvisibility=hidden --param=ssp-buffer-size=4 -fno-omit-frame-pointer -Werror=shadow -include config.h -fPIC -pthread -MD -MQ src/machine/libmachine-core.a.p/machined-core.c.o -MF src/machine/libmachine-core.a.p/machined-core.c.o.d -o src/machine/libmachine-core.a.p/machined-core.c.o -c ../src/machine/machined-core.c
In file included from ../src/basic/memory-util.h:12,
                 from ../src/basic/alloc-util.h:226,
                 from ../src/basic/hash-funcs.h:4,
                 from ../src/basic/hashmap.h:8,
                 from ../src/machine/machined.h:11,
                 from ../src/machine/machined-core.c:3:
../src/fundamental/memory-util-fundamental.h:7:12: fatal error: efi-string.h: No such file or directory
    7 | #  include "efi-string.h"
      |            ^~~~~~~~~~~~~~
```

Which is strange, since the change for `src/fundamental/memory-util-fundamental.h` is in the original commit (https://github.com/redhat-plumbers/systemd-rhel9/commit/493cd5034c3eb091e7163ea1e744a4e07b410710)